### ISSUE TITLE
Does not cache schedule between app restarts

### ISSUE THREAD

**Issue by leto** [fossasia/fossasia-companion-android#64]
To re-created the bug:

- [ ] Open FOSSAsia app, wait for the initial JSON document load
- [ ] View schedule
- [ ] Close app (not just minimize)
- [ ] Open FOSSAsia app again
- [ ] It does the entire schedule reload again, instead of caching locally for say, 1 hr or so

cc @mariobehling

**Comment by batraabhishek** [comment]
@leto Actually this was left intentionally for testing the app as soon as the data is updated. Data source structure was a bit weak, so even a small human error could make the app crash/or prohibiting data parsing. Somehow fixing this was left undone :P :)

**Comment by Vandan-V-Phadke** [comment]
Can we not store it in the sqlite db ?

**Comment by leto** [comment]
@batraabhishek I understand it is useful for testing, but it is a bug that makes a user wait for the entire schedule to load on every start. That was 5-10 seconds on my phone, which is unpleasant. It sounds like it would be nice to have a "refresh schedule" button in the UI somewhere, so the conference guest can decide to update the schedule. Also, this is best done in a background thread, so it does not block the main UI.

**Comment by batraabhishek** [comment]
@leto Yup, It shouldn't be there in the first place. I know the conference is over, but I will quickly fix it for now. As a later step we can implement a background data loading. 

@Vandan-V-Phadke In the current system, there is a versioning system. Initially -1 is stored as version in shared preferences, if the version returned by server is greater then devise stored version, data is downloaded and new version is stored. But we are fetching, but not comparing version in the current build, so it is downloaded every time. :)

**Comment by batraabhishek** [comment]
@Vandan-V-Phadke After parsing the data, we are storing it in sqlite only.
